code/modules/mob/living/simple_animal/simple_animal.dm

```diff
@@ -273,9 +271,10 @@
 /mob/living/simple_animal/gib()
 	if(icon_gib)
 		flick(icon_gib, src)
-	if(meat_amount && meat_type)
-		for(var/i = 0; i < meat_amount; i++)
-			new meat_type(src.loc)
+	if(butcher_results)
+		for(var/path in butcher_results)
+			for(var/i = 1; i <= butcher_results[path];i++)

```

Comment:
it shouldn't i will double check cause of how the harvest code works, hold on i will point you at it